Sources/DatadogObjc/RUMMonitor+objc.swift

```diff
@@ -0,0 +1,221 @@
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import UIKit
+import class Datadog.DDRUMMonitor
+import class Datadog.RUMMonitor
+import enum Datadog.RUMErrorSource
+import enum Datadog.RUMUserActionType
+import struct Datadog.RUMView
+import protocol Datadog.UIKitRUMViewsPredicate
+
+internal struct UIKitRUMViewsPredicateBridge: UIKitRUMViewsPredicate {
+    let objcPredicate: DDUIKitRUMViewsPredicate
+
+    func rumView(for viewController: UIViewController) -> RUMView? {
+        return objcPredicate.rumView(for: viewController)?.swiftView
+    }
+}
+
+@objcMembers
+public class DDRUMView: NSObject {
+    let swiftView: RUMView
+
+    public var path: String { swiftView.path }
+    public var attributes: [String: Any] { swiftView.attributes }
+
+    /// Initializes the RUM View description.
+    /// - Parameters:
+    ///   - path: the RUM View path, appearing as "PATH" in RUM Explorer.
+    ///   - attributes: additional attributes to associate with the RUM View.
+    public init(path: String, attributes: [String: Any]) {
+        swiftView = RUMView(
+            path: path,
+            attributes: castAttributesToSwift(attributes)
+        )
+    }
+}
+
+@objc
+public protocol DDUIKitRUMViewsPredicate: AnyObject {
+    /// The predicate deciding if the RUM View should be started or ended for given instance of the `UIViewController`.
+    /// - Parameter viewController: an instance of the view controller noticed by the SDK.
+    /// - Returns: RUM View parameters if received view controller should start/end the RUM View, `nil` otherwise.
+    func rumView(for viewController: UIViewController) -> DDRUMView?
+}
+
+@objc
+public enum DDRUMErrorSource: Int {
+    /// Error originated in the source code.
+    case source
+    /// Error originated in the network layer.
+    case network
+    /// Error originated in a webview.
+    case webview
+    /// Custom error source.
+    case custom
+
+    internal var swiftType: RUMErrorSource {
+        switch self {
+        case .source: return .source
+        case .network: return .network
+        case .webview: return .webview
+        case .custom: return .custom
+        default: return .custom
+        }
+    }
+}
+
+@objc
+public enum DDRUMUserActionType: Int {
+    case tap
+    case scroll
+    case swipe
+    case custom
+
+    internal var swiftType: RUMUserActionType {
+        switch self {
+        case .tap: return .tap
+        case .scroll: return .scroll
+        case .swipe: return .swipe
+        case .custom: return .custom
+        default: return .custom
+        }
+    }
+}
+
+internal let noopRUMMonitor = DatadogObjc.DDRUMMonitor(swiftRUMMonitor: nil)
+
+@objcMembers
+public class DDRUMMonitor: NSObject {
+    // MARK: - Internal
+
+    internal let swiftRUMMonitor: Datadog.DDRUMMonitor?
```

Comment:
we don't have any use case with `swiftRUMMonitor == nil`, we have `NoOpRUMMonitor` instead.

```swift
internal let noopRUMMonitor = DatadogObjc.DDRUMMonitor(swiftRUMMonitor: nil)
```
the line above doesn't need to exist, we can do `DDRUMMonitor(swiftRUMMonitor: NoOpMonitor())`

Reply:
`NoOpRUMMonitor` is an internal type in Datadog (Swift) module. It can't be used here in DatadogObjc (Objc) module. To provide no-op implementation of `DatadogObjc.DDRUMMonitor` we just create it without providing Swift monitor. All calls to it result with no operation.